hydra_agent/agent.py

```diff
@@ -140,5 +155,55 @@ def process_embedded(self, embedded_resources: list) -> None:
                 embedded_resource['parent_type'],
                 embedded_resource['embedded_url'])
 
+    # Below are the functions that are reponsible to process Socket Events
+    def on_connect(self) -> None:
+        """Method executed when the Agent is successfuly connected to the Server
+        """
+        logger.info('Socket Connection Established - Synchronization ON')
+        self.modifications_table = super().get(self.entrypoint_url_temp +
+                                              '/modification-table-diff').json()
+        if self.modifications_table:
+            self.last_job_id = self.modifications_table[0]['job_id']
+
+    def on_disconnect(self):
+        """Method executed when the Agent is disconnected
+        """
+        pass
+
+    def on_update(self, data) -> None:
+        """Method executed when the Agent receives an event named 'update'
+        This is sent to all clients connected the server under the designed Namespace
+        """
+        new_rows = super().get(self.entrypoint_url_temp +
```

Comment:
I don't see the `data` parameter being used here. I think here we should primarily use the information provided in `data` to update the client instead of using a separate request to the server on `modification-table`. 
What I propose is
1. `data` parameter will contain an object with fields `last_job_id`, `new_job_id`, `operation_type` and `object_uri`. 
2. If `last_job_id` of client and `last_job_id` field of `data` matches then you can directly use `operation_type` and `object_uri` to make appropriate changes at the client and update client's `last_job_id` with `new_job_id`.
3. If they don't match then only client sends a request at `modification-table`.

Reply:
Well, that's much better actually. I agree we should proceed like this.

At first, I was avoiding using the data in the socket afraid that multiple clients would lead to tables having a different order(integrity problem), but putting "last_job_id" inside the socket data event will save a lot of requests and will fall into modification-table-diff only in eventual ocasions when necessary. Good solution! @vddesai1871  :+1: